sunpy/map/map_factory.py

```diff
@@ -207,13 +207,13 @@ def _parse_args(self, *args, **kwargs):
             elif (isinstance(arg, str) and
                   os.path.isdir(os.path.expanduser(arg))):
                 path = os.path.expanduser(arg)
-                files = [os.path.join(path, elem) for elem in os.listdir(path)]
+                files = sorted([os.path.join(path, elem) for elem in os.listdir(path)])
```

Comment:
We should probably refactor this to use pathlib? i.e.

```suggestion
                files = sorted(list(Path(path).glob("*")))
```

Reply:
Sure, but currently `_read_file` does not support `pathlib.Path` objects...

Reply:
Can always cast these to a string of course, before passing them, but I wonder if we should support pathlib at a lower I/O level?

Reply:
Yeah we should totally make `_read_file` support Path objects.

Reply:
Do we also want`Map` to support being passed `pathlib.Path`s? This could make some things a bit cleaner, though it may make the input parsing code a bit messy as we would have to check for either `str` or `pathlib.Path` in the case of directories, filenames, and globs.

Reply:
I am surprised it doesn't already so yes?!

Reply:
This is a bit trickier than I thought...So much of the I/O stuff relies on explicitly passing strings (e.g. in `sunpy.io.file_tools.read_file`, the filepath is pulled apart to check the extension).